app/src/main/java/com/hedvig/app/feature/embark/ui/PreviousInsurerBottomSheet.kt

```diff
@@ -0,0 +1,48 @@
+package com.hedvig.app.feature.embark.ui
+
+import android.graphics.drawable.PictureDrawable
+import android.os.Bundle
+import android.view.View
+import androidx.core.os.bundleOf
+import androidx.fragment.app.setFragmentResult
+import com.bumptech.glide.RequestBuilder
+import com.hedvig.app.feature.embark.passages.previousinsurer.PreviousInsurerAdapter
+import com.hedvig.app.feature.embark.passages.previousinsurer.PreviousInsurerFragment
+import com.hedvig.app.feature.embark.passages.previousinsurer.PreviousInsurerParameter
+import com.hedvig.app.ui.view.ExpandableBottomSheet
+import org.koin.android.ext.android.inject
+
+class PreviousInsurerBottomSheet : ExpandableBottomSheet() {
+
+    private val requestBuilder: RequestBuilder<PictureDrawable> by inject()
+
+    private val insurers by lazy {
+        requireArguments()
+            .getParcelableArrayList<PreviousInsurerParameter.PreviousInsurer>(PREVIOUS_INSURERS)
+            ?: throw IllegalArgumentException("No argument passed to ${this.javaClass.name}")
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.recycler.adapter = PreviousInsurerAdapter(insurers, requestBuilder, ::onInsurerSelected)
+    }
+
+    private fun onInsurerSelected(id: String) {
+        val bundle = Bundle().apply {
+            putString(EXTRA_INSURER_ID, id)
+        }
```

Comment:
Not a dealbreaker, but a cool trick that I think you will enjoy:

```suggestion
		import androidx.os.bundleOf
        val bundle = bundleOf(
        	EXTRA_INSURER_ID to id
        )
```

`core-ktx` is love, `core-ktx` is life 😍

Reply:
Ah nice! Fixing :)